CLONE_RESTORE.md

```diff
@@ -0,0 +1,200 @@
+   4. At this point, the pods will fail, reference problem 2 under the "Challenges faced while implementing this solution with Harbor Operator" to resolve it.
+   5. Delete the `harbor-operator-postgres-operator-*` pod in the `harbor-operator-ns` namespace and wait for the pods to be recreated.
+
+To confirm if backups are being created;
+1. exec into one of the postgres pods e.g.
+   `kubectl exec postgresql-harbor-cluster-harbor-cluster-0 -n harbor-cluster -it --sh`
+2. In the pod, run `envdir /run/etc/wal-e.d/env wal-g backup-list` you'll see a list of backups created
+
+<h2> Restoring backups from s3 using the clone method </h2>
+Although using an in-cluster database configuration that Harbor operator manages has its advantages, it also comes with a few disadvantages, a major one being some of the configurable
+parameters in the postgres CR are hard coded in the Harbor Operator, a few examples are the `patroni` and `clone`parameters (more on these later).
+
+We don't have direct access to the `clone` parameter in the postgres operator, e.g
+``` 
+clone:
+    uid: ""
+    cluster: ""
+```
+We have to find a way to simulate what this parameter does by passing in the needed parameters in the `pod-config` configmap. These are the parameters needed to trigger the clone scripts 
+on postgres pod startup
+
+
+PS: This is the ideal solution for the restoring a base backup as they were tested with regular postgres clusters.
+
+In order to restore backups from s3 these are the steps to follow:
+1. The password to the old cluster needs to be known and stored immediately it is created (This password is generated by the postgres operator)
+   `k get secret standby.<old-cluster-name>.credentials -n <namespace harbor CR was deployed in> -oyaml > secret.yaml`
+   The "creationTimestamp", "ownerReferences", "resourceVersion" and "uid" sections can be removed from the yaml file.
+2. Include the following parameters to the `pod-config` configmap used to create the initial cluster to tell the postgres operator to restore from an existing backup.
+   
+   Note that if you want to also want the restored database to also create backups, you'll have to add the referenced parameters in step 4 of `Creating and storing backups in S3 using WALG`
+   ```
+   CLONE_SCOPE: "" # The name of the cluster you want to clone, ideally it is the folder name within the spilo folder in s3
+   CLONE_METHOD: "CLONE_WITH_WALE"
+   CLONE_HOST: "" # The same value as CLONE_SCOPE
+   CLONE_USER: "postgres"
+   CLONE_WAL_S3_BUCKET: "" # s3 bucket name the backup is stored in
+   CLONE_WAL_BUCKET_SCOPE_SUFFIX: "" # This is the id of the cluster, usually in the clone_scope folder in s3 (s3://<bucket-name>/spilo/<clone-scope>/<scope-suffix>) and it has to start with "/" e.g "/99e85b8a-3f92-4c86-8d42-5056348c1e7e"
+   CLONE_PORT: "5432"
+   CLONE_USE_WALG_RESTORE: "true"
+   USEWALG_RESTORE: "true"
+   CLONE_AWS_REGION: ""
+   CLONE_AWS_ACCESS_KEY_ID:
+   CLONE_AWS_SECRET_ACCESS_KEY:
```

Comment:
It doesn't look like we can set this using an alternative method right now, but does seem like we could make some changes upstream to do so as the postgres-operator accepts [clone parameters](https://github.com/zalando/postgres-operator/blob/920f3dee3e4fa0fdf244a12312a219061ed38ba8/charts/postgres-operator/crds/postgresqls.yaml#L110-L135), which could be configured in the [postgres-spec](https://github.com/goharbor/harbor-operator/blob/master/pkg/cluster/controllers/database/generate.go#L54-L71)